src/node/routes/login.ts

```diff
@@ -98,4 +88,35 @@ router.post("/", async (req, res) => {
   } catch (error) {
     res.send(await getRoot(req, error))
   }
+}
+
+export const router = Router()
+
+router.use((req : any, res : any, next : any) => {
+  const to = (typeof req.query.to === "string" && req.query.to) || "/"
+  if (authenticated(req)) {
+    return redirect(req, res, to, {
+      to: undefined, 
+      password: undefined,
+      pass: undefined,
+    })
+  }
+  next()
 })
+
+router.get("/", async (req : any, res : any) => {
+  if (req.args["enable-get-requests"]) {
+    // `?password` overrides `?pass`
+    if (req.query.password) {
+      return await login(req, res, req.query.password as string)
+    }
+    else if (req.query.pass) {
+      return await login(req, res, req.query.pass as string)
```

Comment:
Minor linter error here due to the newline. Same comment here about typecasting, too.

```suggestion
} else if (req.query.pass && typeof req.query.pass === "string") {
      return await login(req, res, req.query.pass)
```